x/crosschain/keeper/cctx_counter.go

```diff
@@ -0,0 +1,61 @@
+package keeper
+
+import (
+	"cosmossdk.io/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/zeta-chain/node/x/crosschain/types"
+)
+
+func (k Keeper) getCounterValueStore(ctx sdk.Context) prefix.Store {
+	return prefix.NewStore(ctx.KVStore(k.storeKey), []byte(types.CounterValueKey))
+}
+
+func (k Keeper) getCounterIndexStore(ctx sdk.Context) prefix.Store {
+	return prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefix(types.CounterIndexKey))
+}
+
+// GetCctxCounter retrieves the current counter value
+//
+// Will return 0 if the counter has never been set
+func (k Keeper) GetCctxCounter(ctx sdk.Context) uint64 {
+	store := k.getCounterValueStore(ctx)
+	storedCounter := store.Get([]byte(types.CounterValueKey))
+
+	return sdk.BigEndianToUint64(storedCounter)
+}
+
+// SetCctxCounter updates the current counter value
+//
+// This should only be used in setCctxCounterIndex and in state import
+func (k Keeper) SetCctxCounter(ctx sdk.Context, val uint64) {
+	store := k.getCounterValueStore(ctx)
+	store.Set([]byte(types.CounterValueKey), sdk.Uint64ToBigEndian(val))
+}
+
+// getNextCctxCounter retrieves and increments the counter for ordering
+func (k Keeper) getNextCctxCounter(ctx sdk.Context) uint64 {
```

Comment:
I see that this function ( and some other provate functions in this file ) are already covered under the unit tests for `SetCrossChainTx`
But still consider adding some unit tests for these functions

Reply:
So to test this we'd have to export the function right? Since the tests are in a separate package?

Reply:
We have a file to do that, `export_private_functions_test` .